nvtabular/torch_dataloader.py

```diff
@@ -218,20 +222,75 @@ def __len__(self):
         return self.num_chunks
 
     def __iter__(self):
-        buff = queue.Queue(1)
-        threading.Thread(target=self.load_chunk, args=(buff,)).start()
-        for _ in range(self.num_chunks):
-            chunk = buff.get()
-            yield from TensorItr(
-                chunk, batch_size=self.batch_size,
-            )
-
-    def load_chunk(self, out):
         for chunk in self.itr:
-            chunk = create_tensors_plain(
-                chunk, cat_cols=self.cat_cols, cont_cols=self.cont_cols, label_cols=self.label_cols
-            )
-            out.put(chunk)
+            yield chunk
+
+
+class ChunkQueue:
+    def __init__(
+        self,
+        num_chunks=2,
+        batch_size=1,
+        shuffle=False,
+        cat_cols=None,
+        cont_cols=None,
+        label_cols=None,
+    ):
+        self.num_chunks = num_chunks
+        self.batch_size = batch_size
+        self.q_in = queue.Queue(num_chunks)
+        self.q_out = queue.Queue(1)
+        self.cat_cols = cat_cols
+        self.cont_cols = cont_cols
+        self.label_cols = label_cols
+        self.shuffle = shuffle
+
+    def get(self):
+        return self.q_out.get()
+
+    def put(self, obj):
+        # check first for "end"
+        if isinstance(obj, str):
+            # clear the buffer
+            self.create_chunk(final=True)
+            # send bug out
```

Comment:
What's meant by "send bug out" ?

Reply:
Sorry bug out is a term I used in the military, that mean exit or leave.

Reply:
mean exit or leave... sorry this is my old military life spilling over.